nautobot/ipam/models.py

```diff
@@ -772,7 +772,7 @@ class IPAddress(PrimaryModel, StatusModel):
     )
     assigned_object_id = models.UUIDField(blank=True, null=True)
     assigned_object = GenericForeignKey(ct_field="assigned_object_type", fk_field="assigned_object_id")
-    nat_inside = models.OneToOneField(
+    nat_inside = models.ForeignKey(
         to="self",
         on_delete=models.SET_NULL,
         related_name="nat_outside",
```

Comment:
Similarly, this is technically a breaking change, as `IPAddress.nat_outside` is no longer an `IPAddress` object, but is instead a `RelatedManager` object. To avoid breaking existing users of the ORM (i.e., Jobs, plugins), we might need to do something tricky here so that the related_name is `nat_outside_list` and `nat_outside` is a special-purpose property that gets/sets `nat_outside_list.first()`?

Reply:
I agree with this and went ahead and implemented it similar to what you suggest, if not exactly.

Then when we release 2.0, we can simply change the related_name and remove the property, and a few other things for serializers, etc. but pretty straight forward. Let me know if I missed anything.